src/session.rs

```diff
@@ -48,6 +54,21 @@ impl AsRef<Option<Session>> for z_session_t {
     }
 }
 
+impl From<z_session_t> for &'static z_owned_session_t {
```

Comment:
I'd rather have an `AsRef` here, rather than extending the lifetime to static, mostly to avoid accidentally storing something tied to the provided `z_session_t`

Reply:
z_session_t can have any lifetime, but z_owned_session_t is supposed to be always static. Look for example at z_declare_queryable - it explicitly performs std::mem::transmute::<&Session, &'static Session>(s). Or to z_declare_subscriber - it creates z_owned_subscriber_t without any lifetimes related to session provided. All such functions contains 'unsafe' blocks which implicitly or explicitly converts 'z_session_t' into &'static Sesssion. I've just made this conversion explicit in the definition of z_owned_session_t itself.